### DISCUSSION TITLE
Is symmetric encryption in kryptor authenticated?

### DISCUSSION THREAD

**Discussion by ahnedeee** [samuel-lucas6/Kryptor#66] [Q&A] [answered]
Is symmetric encryption in kryptor authenticated?
Will there be an error thrown if the symmetrically encrypted file has been tampered with?

**Comment by samuel-lucas6** [comment, marked as answer]
> Is symmetric encryption in kryptor authenticated?

Yep, ChaCha20-Poly1305 is used. You can find more detail [here](https://www.kryptor.co.uk/specification).

> Will there be an error thrown if the symmetrically encrypted file has been tampered with?

Yes, decryption will fail with an error. Even if the file is modified in the last chunk, the rest of the decrypted file will be deleted from disk.

**Reply by ahnedeee** [reply]
Thanks for the response
